src/components/StructureExplorer/MoleculeList.tsx

```diff
@@ -0,0 +1,204 @@
+
+interface FilterProps {
+  filter: string;
+  setFilter: React.Dispatch<React.SetStateAction<string>>;
+}
+
+interface MoleculesProps {
+  molecules: IMolecule[];
+}
+interface MoleculeListProps extends MoleculesProps {
+  idCode: string;
+  search: SearchType;
+  db: MoleculesDB;
+}
+interface CellProps {
+  columnIndex: number;
+  rowIndex: number;
+  style: CSSProperties;
+}
+
+function Filter({ filter, setFilter }: FilterProps): JSX.Element {
+  return (
+    <div className="flex items-center space-x-2">
+      <div className="text-[#0A4E7A]">Filter by name :</div>
+      <div className="flex h-6 w-36 items-center rounded-lg bg-white p-1 text-sm">
+        <input
+          type="text"
+          name="filter"
+          className="h-full w-full px-2 font-normal focus:outline-none"
+          value={filter}
+          onChange={(e) => {
+            setFilter(e.target.value);
+          }}
+        />
+        <MdClose
+          className="cursor-pointer"
+          onClick={() => {
+            setFilter('');
+          }}
+        />
+      </div>
+    </div>
+  );
+}
+
+const cell =
+  (molecules: IMolecule[]) =>
+  ({ columnIndex, rowIndex, style }: CellProps) => {
+    if (columnIndex + rowIndex * 3 < molecules.length) {
+      const mol = molecules[columnIndex + rowIndex * 3];
+      return (
+        <div style={style} className="overflow-x-hidden">
+          {mol && <MoleculeInfo mol={mol} />}
+        </div>
+      );
+    }
+    return null;
+  };
+
+function dbToMolecules(moleculeDb: any): IMolecule[] {
+  const molecules: IMolecule[] = [];
+  for (const entry of moleculeDb) {
+    const data = entry.data;
+    molecules.push({
+      id: data.id,
+      code: data.code,
+      smiles: data.smiles,
+      mf: { type: 'string', value: data.mf },
+      em: data.em,
+      mw: entry.properties.mw,
+      actID: { type: 'string', value: entry.idCode },
+      // eslint-disable-next-line camelcase
+      act_idx: entry.index,
+    });
+  }
+  return molecules;
+}
+
+function Molecules({ molecules }: MoleculesProps): JSX.Element {
+  const { selectedTitle, setSelectedTitle } = useIdContext();
+
+  useEffect(() => {
+    if (selectedTitle === '') {
+      setSelectedTitle(molecules[0]?.code);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const BREAKPOINT1 = 1100;
+  const BREAKPOINT2 = 1024;
+  const BREAKPOINT3 = 700;
+  const [colItemsCount, setColItemsCount] = useState(
+    (window.innerWidth <= BREAKPOINT1 && window.innerWidth > BREAKPOINT2) ||
+      window.innerWidth <= BREAKPOINT3
+      ? 2
+      : 3,
+  );
+  useEffect(() => {
+    const handleResize = () =>
+      setColItemsCount(
+        (window.innerWidth <= BREAKPOINT1 && window.innerWidth > BREAKPOINT2) ||
+          window.innerWidth <= BREAKPOINT3
+          ? 2
+          : 3,
+      );
+    window.addEventListener('resize', handleResize);
+    return () => {
+      window.removeEventListener('resize', handleResize);
```

Comment:
You could use a resize observer instead: https://www.npmjs.com/package/use-resize-observer